modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCppCodegen.java

```diff
@@ -418,13 +418,10 @@ private void addForwardDeclarations(CodegenModel parentModel, Map<String, Object
                 if( !childPropertyType.equals(childModel.classname) || childPropertyType.equals(parentModel.classname) || !childModel.hasVars ){
```

Comment:
What is the `!childModel.hasVars` check good for? 
To me it looks like we might want to get rid of that too